mvp_cd/deploy_app/views.py

```diff
@@ -1,118 +1,159 @@
 from rest_framework.decorators import api_view
 from rest_framework import status
 from rest_framework.response import Response
-from mvp_cd.deploy_app.models import StagingBuildInfo
-from mvp_cd.deploy_app.models import ProductionBuildInfo
-from mvp_cd.deploy_app.models import ProductionWorkerBuildInfo
-from mvp_cd.deploy_app.models import StagingWorkerBuildInfo
+from mvp_cd.deploy_app.models import (BuildInfo, ServerBuildInfo, ServerType)
 import subprocess
 from django.shortcuts import render_to_response
 import json
+from django.shortcuts import get_object_or_404
 
 
-# Create your views here.
+ALLOWED_BRANCHES = ['staging', 'production']
 
 
 @api_view(['POST'])
-def deploy(request):
+def deploy_view(request):
     if request.data['payload']['outcome'] == 'success':
+        if request.data['payload']['branch'] not in ALLOWED_BRANCHES:
+            return Response(status=status.HTTP_204_NO_CONTENT)
+
+        vcs_revision = request.data['payload']['vcs_revision']
```

Comment:
Incoming data needs to be validated for type and security. User data can't be trusted.